community/kernel/src/main/java/org/neo4j/kernel/impl/transaction/log/BatchingTransactionAppender.java

```diff
@@ -389,7 +390,19 @@ public void force() throws IOException
         synchronized ( logFile )
         {
             writer.emptyBufferIntoChannelAndClearIt();
+        }
+        // Force the writer outside of the lock.
+        // This allows multiple threads forcing at the same time to piggy-back onto one another.
```

Comment:
Good point. It's correct that closing a channel does not imply a force. I think we have to force the old log before we close it and create a new log file, such that we ensure that writes to old log file are ordered before any writes to the new log file. I think most file systems will order the creation of a file before any writes to that file (not sure what the effect would be if they didn't) but this ordering doesn't say anything about prior writes to existing files.